src/main/java/net/sf/mzmine/modules/peaklistmethods/identification/formulapredictionpeaklist/FormulaPredictionPeakListTask.java

```diff
@@ -163,18 +163,44 @@ public void run() {
           massRange.upperEndpoint(), elementCounts);
 
       IMolecularFormula cdkFormula;
+
+      // create a map to store ResultFormula and relative mass deviation for sorting
+      Map<Double, ResultFormula> possibleFormulas = new HashMap<Double, ResultFormula>();

```

Comment:
You can use a TreeMap here and keep all the results sorted. There is no need to create a HashMap and then convert it into a TreeMap later.

Reply:
agreed